tutorials/intro-to-swagger-api.md

```diff
@@ -0,0 +1,109 @@
+
+## Background
+Our API is public, it is always theoretically possible for a user to bypass any UI controls we set up by hitting the endpoints directly or via tools.  The UI has restrictions as well, but ultimately its purpose is as a more friendly interface to the API rather than a security mechanism in itself.
+
+For more details on our API endpoints see the [README](https://github.com/all-of-us/workbench) and the [API Code Structure](https://github.com/all-of-us/workbench/blob/master/api/docs/code-structure.md).
+
+Note that we often refer to the AoU RW's back end GAE application as "the api" but we also use "the api" to refer to the collection of endpoints we expose.  It can get tricky!  We really should think of a real name for this app.
+
+Endpoints are defined in the Swagger/OpenAPI YAML format.  We run code generation tools as part of the standard compilation and deployment processes to create client code in Java and Typescript (for the UI) as well as server interface code in Java for us to implement with the app.
+
+## Caveats / Troubleshooting
+
+1. The build directory may have some empty directories that are needed for the build to succeed
+2. You may need to restart the local server for your changes to take effect
+3. It can be difficult to detect syntax errors in your YAML file. Use a validator to ensure that you do not have errors
+
+## Tasks
+
+### (API-SWAG-1) Compile The API 
+
+1. In the API directory run: 
+    `./gradlew clean compileJava`
+    This will generate an error (I am not sure why)
+    Run `./gradlew compileJava` again. This should run to completion.
+
+    It will run the `compileGeneratedJava` task, creating and compiling the code generated by swagger and it will compile our hand written java code
+
+    Some additional things you can do to explore the code (in the `api/` directory):
+    1. Look in the `build`  directory.
+       1. Identify where the generated code is
+       2. Identify where some of the generated code is being used in the application
+    2. Run `clean` and examine the build directory
+       1. What happens?
+       2. Re-create the generated code
+
+### (API-SWAG-2) Add/Remove A Parameter From An Endpoint 
+In the previous task we generated the swagger java code and compile our handwritten code.
+Let's explore some of our handwritten code and see how it interacts with the Swagger/OpenAPI code
+1. Go to `src/main/java/org/pmiops/workbench/api/WorkspacesController.java`
+   1. Let's look at one of the overriden methods in this file - `getWorkspace`
+   2. Add a parameter to this method. Try compiling - what do you think will happen when you compile?
+   3. Did your IDE respond to the change? (If not check your IDE settings)
+2. Undo your changes above, this time remove a parameter from the method.
+
+### (API-SWAG-3) Bring up the server and inspect an endpoint
```

Comment:
Note: http://localhost:8081/v1/config is an easy one to check, since it doesn't require auth. It may be worth showing this before moving onto an authenticated call, which is a more daunting task.